draft-ietf-rats-corim.md

```diff
@@ -1066,6 +1066,182 @@ The following describes each member of the `concise-bom-tag` map.
+In particular, it is expected that the cost associated with the initialisation phase can be amortised across multiple appraisals.
+Implementers are free to do as they please, as long as the stated invariants are not broken.
+
+In the following, if a MUST-level requirement is violated, the entire procedure is aborted.
+
+## Appraisal Context initialisation
+
+The goal of the initialisation phase is to load the CoRIM Appraisal Context with objects such as CoRIM tags, cryptographic validation key material (e.g., raw public keys, root certificates, intermediate CA certificate chains, etc.) etc. that will be used in the subsequent Evidence Appraisal phase.
+
+### CoRIM Selection
+
+All available Concise Reference Integrity Manifests (CoRIMs) are collected. A verifier may be pre-configured with a large number of tags describing many types of device. All CoRIMs are loaded at this stage, later stages will select the CoRIMs appropriate to the evidence appraisal step.
+
+CoRIMs that are not within their validity period, or that cannot be associated with an authenticated and authorised source MUST be discarded.
+
+CoRIM that are secured by a cryptographic mechanism such as a signature MUST be discarded if the signature chain does not validate successfully.
+
+Other selection criteria MAY be applied.
+For example, if the Evidence format is known in advance, CoRIMs that do not match the expected profile can be readily discarded.
+
+The selection process MUST yield at least one usable CoRIM.
+
+### CoBOMs Extraction
+
+All the available Concise Bill Of Material tags (CoBOMs) are collected from the selected CoRIMs.
+
+If any CoBOM is present then other tags in the CoRIM which are not activated by the CoBOM MUST be ignored.
+
+### Tags Identification and Validation
+
+Potentially valid tags -- i.e., Concise Module Identifiers (CoMID) and/or Concise Software Identifiers (CoSWID) and/or Concise Trust Anchor Stores (CoTS) -- are chosen from the selected CoRIMs.
+
+All the extracted tags MUST be successfully located and be syntactically and semantically valid.
+In particular, any internal cross-reference (e.g., CoMID-CoSWID linking triples) MUST be successfully resolved.
+
+### Appraisal Context Construction
+
+All of the potential tags are loaded into the Appraisal Context. The complete contents of each tag are loaded together, including Reference Values, Endorsed Values and cryptographic verification key material.
+
+This concludes the initialisation phase.
+
+## Evidence Collection
+
+In the evidence collection phase the verifier communicates with attesters to collect evidence. 
+
+The first part of the evidence collection phase does not perform any crypographic validation. This allows verifiers to use untrusted code for their initial evidence collection.
+
+The results of the evidence collection are protocol specific data and transcripts which can be processed by the verifier.
+
+### Cryptographic validation of Evidence
```

Comment:
The RATS Architecture refers to this as 'appraisal'. We should use that term unless we're describing something else.